nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/statuses/StatusServlet.java

```diff
@@ -55,18 +61,32 @@
 
     public static final String PARAM_RELOAD = "reload";
 
+    public static final String PROBE_PARAM = "probe";
+
     private AbstractRuntimeService runtimeService;
 
+    protected ProbeManager pm;
+
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         String param = req.getParameter(PARAM);
         if (param != null) {
             doPost(req, resp);
         } else {
-            sendResponse(resp, "Ok");
+            HealthCheckResult result = getOrRunHealthCheck(null);
+            sendHealthCheckResponse(resp, result);
         }
     }
 
+    protected void sendHealthCheckResponse(HttpServletResponse resp, HealthCheckResult result) throws IOException {
+        if (result.healthy) {
+            resp.setStatus(HttpServletResponse.SC_OK);
+        } else {
+            resp.setStatus(HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
You should use `HttpServletResponse.SC_...` everywhere, no need for `HttpStatus` here.